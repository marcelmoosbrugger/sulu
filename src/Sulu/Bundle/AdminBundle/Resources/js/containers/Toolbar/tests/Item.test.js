/* eslint-disable flowtype/require-valid-file-annotation */
import {render, shallow} from 'enzyme';
import Item from '../Item';
import React from 'react';

test('Render item', () => {
    expect(render(<Item />)).toMatchSnapshot();
});

test('Render disabled item', () => {
    expect(render(<Item enabled={false} />)).toMatchSnapshot();
});

test('Click on item fires onClick callback', () => {
    const clickSpy = jest.fn();
    const item = shallow(<Item onClick={clickSpy} />);

    item.simulate('click');

    expect(clickSpy).toBeCalled();
});

test('Click on item does not fire onClick callback if item is disabled', () => {
    const clickSpy = jest.fn();
    const item = shallow(<Item onClick={clickSpy} enabled={false} />);

    item.simulate('click');

    expect(clickSpy).toHaveBeenCalledTimes(0);
});
