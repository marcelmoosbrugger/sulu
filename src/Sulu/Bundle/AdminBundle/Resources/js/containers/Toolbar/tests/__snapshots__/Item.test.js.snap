// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render disabled item 1`] = `
<button
  class="item"
  disabled=""
>
  <span
    aria-hidden="true"
    class="icon fa fa-undefined"
  />
  <span
    class="title"
  />
</button>
`;

exports[`Render item 1`] = `
<button
  class="item"
>
  <span
    aria-hidden="true"
    class="icon fa fa-undefined"
  />
  <span
    class="title"
  />
</button>
`;
